couchdb/server/ddoc.py

```diff
@@ -58,3 +58,52 @@ def add_ddoc(self, server, ddoc_id, ddoc):
         log.debug('Cache design document `%s`', ddoc_id)
         self.cache[ddoc_id] = ddoc
         return True
+
+    def run_ddoc_func(self, server, ddoc_id, fun_path, fun_args):
+        """
+        :param server: Query server instance.
+        :type server: :class:`~couchdb.server.BaseQueryServer`
+
+        :param ddoc_id: Design document id, holder of requested function.
+        :type ddoc_id: unicode
+
+        :param fun_path: List of key by which request function could be found
+                         within ddoc object. First element of this list is
+                         ddoc command.
+        :type fun_path: list
+
+        :param fun_args: List of function arguments.
+        :type fun_args: list
+
+        :return: Result of called design function if any available.
+                 For example, lists doesn't explicitly returns any value.
+
+        .. versionadded:: 0.11.0
+        """
+        ddoc = self.cache.get(ddoc_id)
+        if ddoc is None:
+            msg = 'Uncached design document: {0}'.format(ddoc_id)
+            log.error(msg)
+            raise FatalError('query_protocol_error', msg)
```

Comment:
You should [check](https://github.com/apache/couchdb/blob/master/share/server/loop.js#L85) JavaScript QS instead.